src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -0,0 +1,302 @@
+    "468e88"; "0257d9"; "7b5729"; "beb964";
+  }
+
+local campaignData = {}
+
+local tokenZone = getObjectFromGUID("ef3b5f")
+local deckImporter = getObjectFromGUID("a28140")
+local investigatorCounter = getObjectFromGUID("f182ee")
+
+local campaignBoxGUID = ""
+local traumaValues = false
+local deckIds = false
+local campaignGuidePage = false
+
+function onLoad(save_state)
+    self.createButton({
+        click_function = "findCampaignFromToken",
+        function_owner = self,
+        label = "Import",
+        position = {x=-1, y=0.2, z=0},
+        width = 350,
+        height = 150,
+        scale = {2, 1, 2},
+    })
+    self.createButton({
+        click_function = "createCampaignToken",
+        function_owner = self,
+        label = "Export",
+        position = {x=1, y=0.2, z=0},
+        width = 350,
+        height = 150,
+        scale = {2, 1, 2},
+    })
+end
+
+function findCampaignFromToken(_, _, _)
+    local coin = nil
+    for _, obj in ipairs(tokenZone.getObjects()) do
+        if obj.hasTag("ImporterToken") then
+            coin = obj
+        end 
+    end
+    if coin == nil then
+        broadcastToAll("Could not find importer token", Color.Red)
+    else
+        local importData = JSON.decode(coin.getGMNotes())
+        campaignBoxGUID = importData[1]
+        local campaignBox = getObjectFromGUID(campaignBoxGUID)
+        if campaignBox.type == "Generic" then
+            campaignBox.call("buttonClick_download")
+        end
+        Wait.condition(
+            function()
+                if #campaignBox.getObjects() > 0 then
+                    placeCampaignFromToken(importData)
+                else
+                    createCampaignFromToken(importData)
+                end
+            end,
+            function()
+                local obj = getObjectFromGUID(campaignBoxGUID)
+                if  obj == nil then 
+                    return false 
+                else
+                    return obj.type == "Bag" and obj.getLuaScript() ~= ""
+                end
+            end,
+            2,
+            function() broadcastToAll("Error loading campaign box") end
+        ) 
+    end
+end
+
+function placeCampaignFromToken(importData)
+    getObjectFromGUID(campaignBoxGUID).call("buttonClick_place")
+    Wait.condition(
+        function() createCampaignFromToken(importData) end,
+        function() return findCampaignLog() ~= nil end,
+        2,
+        function() broadcastToAll("Error placing campaign box") end
+    )
+end
+
+function createCampaignFromToken(importData)
+    findCampaignLog().destruct()
+    findChaosBag().destruct()
+    --create campaign log
+    spawnObjectData({data = importData[2]}) 
+    --create chaos bag
+    spawnObjectData({data = importData[3]})
+    --populate trauma values
+    if importData[4] then
+        updateCounters(importData[4])
+    end
+    --populate ArkhamDB deck IDs
+    if importData[5] then
+        deckImporter.call("updateDeckIds", importData[5])
```

Comment:
Avoid call() directly on other objects.  Use the API pattern instead:
https://docs.google.com/document/d/14ZX5JdIXx0rpGQ2HUQk91hWXBOqtZFY1xXDhwDRz4Z0/edit?usp=sharing